pallets/automation-time/src/lib.rs

```diff
@@ -155,11 +157,30 @@ pub mod pallet {
 		}
 	}
 
+	/// The struct that stores data for a missed task.
+	#[derive(Debug, Eq, PartialEq, Encode, Decode, TypeInfo)]
+	#[scale_info(skip_type_params(T))]
+	pub struct MissedTask2<T: Config> {
+		owner_id: T::AccountId,
+		task_id: TaskId<T>,
+		execution_time: UnixTime,
+	}
+
+	impl<T: Config> MissedTask2<T> {
+		pub fn create_missed_task(
+			owner_id: T::AccountId,
+			task_id: TaskId<T>,
+			execution_time: UnixTime,
+		) -> MissedTask2<T> {
+			MissedTask2::<T> { owner_id, task_id, execution_time }
+		}
+	}
+
 	/// The struct that stores all information needed for a task.
 	#[derive(Debug, Eq, Encode, Decode, TypeInfo)]
 	#[scale_info(skip_type_params(T))]
 	pub struct Task<T: Config> {
-		owner_id: AccountOf<T>,
+		pub owner_id: AccountOf<T>,
```

Comment:
Does `owner_id` still make sense as a part of `Task` when the storage structure is:
```
owner_id:
  task_id:
    task: {
      owner_id
      ...
    }
```
Maybe this is outside the scope of this change, but we should atleast consider removing the duplicate owner_id in the future.

Reply:
it does not make sense anymore, but is outside the scope. There is also no reason to keep `provided_id` in the struct that I can think of. I figured we will need to change the task struct soon anyway so can bundle the change then

Reply:
Works for me!